media-plugins/adlplug/adlplug-9999.ebuild

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils multibuild git-r3
+
+DESCRIPTION="FM synthesizer plugins based on OPL3 and OPN2 sound chip emulations"
+HOMEPAGE="https://github.com/jpcima/ADLplug"
+
+EGIT_REPO_URI="https://github.com/jpcima/${PN}.git"
+if [[ ${PV} == *9999 ]]; then
+	KEYWORDS=""
+else
+	# Upstream release tarballs do not contain the necessary third party libs;
+	# these are available via git submodules per upstream build instructions
+	EGIT_COMMIT="v${PV}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Boost-1.0 BSD GPL-3 GPL-2+ LGPL-3 LGPL-2.1 MIT ISC"
+SLOT="0"
+
+# Upstream offers a VST3 build option, but unfortunately JUCE 5.4.1
+# doesn't support VST3 on Linux
+# see https://github.com/WeAreROLI/JUCE/issues/307
+IUSE="+opl3 +opn2 +lv2 +vst jack alsa pch assertions"
+REQUIRED_USE="
+	|| ( opl3 opn2 )
+	|| ( lv2 vst jack alsa )
+"
+
+# JUCE will stop the build if alsa-lib is not present, regardless
+# of the build options
+DEPEND="
+	x11-libs/libXinerama
+	x11-libs/libXext
+	x11-libs/libXrandr
+	x11-libs/libXcursor
+	media-libs/freetype
+	media-libs/alsa-lib
+	jack? ( virtual/jack )
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	dev-util/cmake
+	|| (
+		>=sys-devel/gcc-4.8.1
```

Comment:
The only reason I listed them is because the build requires a compiler with C++11 support (see [here](https://github.com/jpcima/ADLplug/blame/master/README.md#L60)). Those are the minimum versions of GCC and Clang that support it; those lines just specify that one or the other needs to be there. Since they're quite old now, I imagine most users have more recent versions; I just figured it wouldn't hurt to specify since Gentoo is so flexible. If you think it's unnecessary I'm not opposed to taking that part out.

Conversely, I left out icc because when I wrote the ebuild I thought it didn't have full C++11 support. More recently I found out that it does now, at least on Linux. So, we could theoretically support it too here if you think it's worth leaving this part in (although I don't have a copy to test the build with).